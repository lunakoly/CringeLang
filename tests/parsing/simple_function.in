fun test(a: Int = 10, b: String): Float
    a = 10
    b = 29

fun test: Boolean print * 2

fun rest
    keker