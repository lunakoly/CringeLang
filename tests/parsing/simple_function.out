==== Diagnostics ====

==== AST ====
[fun test ([var [a]: Int = [10], var [b]: String]): Float [[([a] = [10]), ([b] = [29])]], fun test ([]): Boolean [[(print * 2)]], fun rest ([]): <!MISSING RETURN TYPE!><!> [[[keker]]]]

==== RESOLVED ====
*** FILE 0 ***
[fun test ([var [a]: Int = [10], var [b]: String]): [ERROR:UNRESOLVED_IDENTIFIER] [[([a] = [10]), ([b] = [29])]], fun test ([]): [ERROR:UNRESOLVED_IDENTIFIER] [[(print * 2)]], fun rest ([]): [ERROR:UNRESOLVED_IDENTIFIER] [[[keker]]]]

==== Done ====
